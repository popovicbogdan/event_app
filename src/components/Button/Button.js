import React from "react";
import PropTypes from "prop-types";
import { ButtonStyled } from "./Button.styled";
import Icon from "../Icon/Icon";

const Button = ({ className, onClick, icon, children }) => {
  return (
    <ButtonStyled className={className} onClick={onClick}>
      {icon && <Icon>{icon}</Icon>}
      {children}
    </ButtonStyled>
  );
};

Button.defaultProps = {
  className: "btn"
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  icon: PropTypes.elementType,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
};

export default React.memo(Button);
