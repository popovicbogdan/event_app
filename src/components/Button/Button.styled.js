import styled from "styled-components";

export const ButtonStyled = styled.button.attrs({
  type: "button"
})`
  display: inline-flex;
  justify-content: center;
  background: #e838ff;
  color: white;
  border: none;

  :hover {
    background: #960c99;
  }
  &:focus {
    background: #850071;
  }
`;
