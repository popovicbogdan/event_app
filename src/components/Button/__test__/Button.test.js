import React from "react";

import Button from "../Button";
import { shallow, configure, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

// jest.mock("../../Icon/Icon", () => "Icon");
configure({ adapter: new Adapter() });
describe("<Button />", () => {
  const mockOnClick = jest.fn();
  const wrapper = shallow(
    <Button onClick={mockOnClick} icon="search">
      Button
    </Button>
  );

  it("should trigger onClick", () => {
    wrapper.simulate("click");
    expect(mockOnClick).toBeCalled();
  });
  it("should render children", () => {
    expect(wrapper.contains("Button")).toEqual(true);
  });
  it("should have default className 'btn'", () => {
    expect(wrapper.hasClass("btn")).toBe(true);
  });
  it("should have an icon and text", () => {
    expect(wrapper.children()).toHaveLength(2);
  });
  it("should render correclty", () => {
    const button = renderer
      .create(
        <Button onClick={mockOnClick} icon="search">
          Text
        </Button>
      )
      .toJSON();
    expect(button).toMatchSnapshot();
  });
});
