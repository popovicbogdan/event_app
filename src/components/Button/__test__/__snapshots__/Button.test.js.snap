// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correclty 1`] = `
<button
  className="sc-AykKC hBViMd btn"
  data-testid="button"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  type="button"
>
  <i
    className="sc-AykKD fZGMjR material-icons"
    data-testid="icon"
  >
    search
  </i>
  Text
</button>
`;
